### ISSUE TITLE
Extract get-wcga-colors

### ISSUE THREAD

**Issue by Kikobeats** [FormidableLabs/react-image-palette#12]
I'm loving what this library does. Unfortunately it is very limited to a React stack.

I want to ask If it make senses for you extract the main logic (WCAG Compliant Colors) in a separate module, easy to use indepent of your project stack.

It also can have the flexibility to use a different extractor color than `ColorThief`.

I'm suggesting in build a separate module since [this line](https://github.com/FormidableLabs/react-image-palette/blob/master/src/get-image-palette.js#L153).

An example of using it in inside `react-image-palette`:

```js
const palette = getColorPalette(image, colorThief)

// palette should be something like [ '#3c0c1c', '#fa4c2c', '#4a272d', '#834a3b' ] 
// no matters how you get the colors
const compliantColors = getContrastPalette(palette)

// It returns an array of object like
// [{ color, score, contrast }, { color, score, contrast }, { color, score, contrast }]

/* the rest of your logic for get backgroundColor, color and alternativeColor */
```

What do you think?

**Comment by aweary** [comment]
`image-palette-core` has been extracted in https://github.com/FormidableLabs/react-image-palette/pull/14
